Patch: 21_newlist_help.patch
Author: Tollef Fog Heen <tfheen@debian.org>
make newlist be a bit friendlier (Closes: #309339, #170362)
Index: b/bin/newlist
===================================================================
--- a/bin/newlist	2010-05-24 17:50:50.000000000 +0200
+++ b/bin/newlist	2010-05-24 17:51:04.000000000 +0200
@@ -88,12 +88,16 @@
 defined in your Defaults.py file or overridden by settings in mm_cfg.py).
 
 Note that listnames are forced to lowercase.
+
+The list admin address need to be a fully-qualified address, like
+owner@example.com, not just owner.
 """
 
 import sys
 import os
 import getpass
 import getopt
+import grp
 
 import paths
 from Mailman import mm_cfg
@@ -122,6 +126,9 @@
 
 
 def main():
+    gid = grp.getgrnam(mm_cfg.MAILMAN_GROUP)[2]
+    if os.getgid() != gid and gid not in os.getgroups():
+        os.setgid(gid)
     try:
         opts, args = getopt.getopt(sys.argv[1:], 'hql:u:e:',
                                    ['help', 'quiet', 'language=',
@@ -204,7 +211,7 @@
         except Errors.BadListNameError, s:
             usage(1, _('Illegal list name: %(s)s'))
         except Errors.EmailAddressError, s:
-            usage(1, _('Bad owner email address: %(s)s'))
+            usage(1, _('Bad owner email address: %(s)s.  Owner addresses need to be fully-qualified names, like "owner@example.com", not just "owner".'))
         except Errors.MMListAlreadyExistsError:
             usage(1, _('List already exists: %(listname)s'))
 
